geo/src/algorithm/simplifyvw.rs

```diff
@@ -290,11 +289,20 @@ where
         // remove it from the simulated "linked list"
         adjacent[smallest.current as usize] = (0, 0);
         counter -= 1;
-        // remove stale segments from R* tree
-        // we have to call this twice because only one segment is returned at a time
-        // this should be OK because a point can only share at most two segments
-        tree.lookup_and_remove(&Point(orig.0[smallest.right]));
-        tree.lookup_and_remove(&Point(orig.0[smallest.left]));
+        // Remove stale segments from R* tree
+        let left_point = Point(orig.0[left as usize]);
+        let middle_point = Point(orig.0[smallest.current]);
+        let right_point = Point(orig.0[right as usize]);
+
+        let line_1 = Line::new(left_point, middle_point);
+        let line_2 = Line::new(middle_point, right_point);
+        assert!(tree.remove(&line_1).is_some());
+        assert!(tree.remove(&line_2).is_some());
```

Comment:
Note that I originally had a different assertion:
`assert!(tree.remove(&line_1).is_some() || tree.remove(&line1.reverse().is_some())`
as I was not sure in which direction a line would have been loaded.

I then realized that, as long as all lines are going "in the same direction", these assumptions should hold: If edges A->B and B->C exist, then they should be loaded as `(A,B), (B,C)` in contrast to `(A,B),(C,B))`.

Since all incoming lines are either the lines of a line string (which is directed) or the inner rings of a polygon (which are also directed), these assertions should hold. However, you may want to check this yourselves. Please note that the spade algorithm was not susceptible to this, that's why I'm so cautious. It was though, at the same time, much more inefficient.

Reply:
As you say, this could only be a problem for us if someone constructs an invalid geometry. @frewsxcv, I know we aren't currently checking for that, but I don't know if this is the right place to enforce it – I think it's reasonable for individual algorithms to assume that incoming geometries are valid.

Reply:
agreed. we still need to work on our validation story in this crate, but for the time being, i think it's okay to assume geometry coming in is invalid so this looks good 👍